File: CSharp/Shared/ClientFactory.cs
using System;
using System.IO;

namespace Shared
{
	public static class ClientFactory
	{
		public static IClientFactory Create(int transferDelay = 0)
		{
			var rootDir = Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "../../../../sftp"));
			return new LocalFileClientFactory(rootDir, "", transferDelay);
		}
	}
}

File: FSharp/Shared/ClientFactory.fs
module ClientFactory

    open SftpClient
    open SshNetClient
    open LocalFileClient

    let createClientFactory () =
        LocalFileClientFactory("sftp", "", 0<s/MB>)

    let createClientFactoryWithTransferDelay (transferDelay) =
        LocalFileClientFactory("sftp", "", transferDelay)
